File: Spring/Money/user/src/main/java/focandlol/dto/UserInfoDto.java
package focandlol.dto;

import focandlol.domain.UserInfo;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.*;

public class UserInfoDto {

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class Request {

        @Schema(description = "UserName", example = "kkm")
        private String userName;
        @Schema(description = "UserRegistrationNum", example = "11111111111")
        private String userRegistrationNumber;
        @Schema(description = "userIncome", example = "1111111111")
        private Long userIncomeAmount;

        public UserInfo toEntity(String key) {
            return UserInfo.builder()
                    .userName(this.userName)
                    .userRegistrationNumber(this.userRegistrationNumber)
                    .userIncomeAmount(this.userIncomeAmount)
                    .userKey(key)
                    .build();

        }
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Response{
        private String userKey;
    }
}

File: Spring/fintech/api/src/main/kotlin/focandlol/api/loan/request/UserInfoDto.kt
package focandlol.api.loan.request

import focandlol.domain.domain.UserInfo

data class UserInfoDto(
    val userKey: String,
    val userName :String,
    val userRegistrationNumber: String,
    val userIncomeAmount: Long)
{
    fun toEntity(): UserInfo =
        UserInfo(userKey, userRegistrationNumber, userName, userIncomeAmount)
}